be/src/exec/vectorized/json_scanner.h

```diff
@@ -134,4 +117,43 @@ class JsonReader {
 #endif
 };
 
+class JsonParser {
+public:
+    JsonParser() = default;
+    virtual ~JsonParser() = default;
+    virtual Status parse(uint8_t* data, size_t len, size_t allocated) = 0;
+    virtual Status get_next(simdjson::ondemand::object* row) = 0;
+};
+
+class JsonDocumentStreamParser : public JsonParser {
+public:
+    Status parse(uint8_t* data, size_t len, size_t allocated) override;
+    Status get_next(simdjson::ondemand::object* row) override;
+
+private:
+    uint8_t* _data;
+    simdjson::ondemand::parser _parser;
+
+    // _first_shot denotes wheather the first element of the document stream would be returned.
+    bool _first_shot = false;
```

Comment:
When the first element do not return?

Reply:
The first element is returned when the `get_next()` is called for the first time after the `parse()`.

Reply:
`_first_shot ` is the flag to tell us weather to forward the iterator.

Reply:
Does the _first_shot is necessary? Could we use the get_next() to forward the iterator automatically?

Reply:
The `get_next()` is divided into 2 parts: `next()` and `get()`, and the `_first_shot ` is removed.